internal/operators/operators.go

```diff
@@ -0,0 +1,301 @@
+// Copyright 2022-2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package operators
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/inspektor-gadget/inspektor-gadget/internal/logger"
+	"github.com/inspektor-gadget/inspektor-gadget/internal/parser"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+)
+
+type Runner interface {
```

Comment:
We actually have one more: `gadgets.Runner`.

Under the hood, they're all the same - implemented by `gadgetrunner.GadgetRunner`, which acts as kind of glue between all components.

Using different interfaces helps to add a scope and avoid circular dependencies.

For example: while `gadgetrunner.GadgetRunner` offers things useful/required for Operators or Runtimes, they are not exposed in the `gadgets.Runner` interface as they aren't needed in the gadgets. So we can always keep the interfaces as small/clean as possible for the given scope they're used in - which I think makes it actually easier to work with in the end.

Reply:
I think what they do is clear through the scope they're in - the operator.Runner does stuff related to operators and so on - "Runner" shows their shared origin.

What would you name it?

/cc @mauriciovasquezbernal @blanquicet

Reply:
Actually .... Now that I swim more through the code, I think it is fine that all of them are called in the same way. However, I am not anymore sure `Runner` is the correct name. Let me think about it.

cc @mauriciovasquezbernal

Reply:
I did some experiments to make the `runner` goal clearer [here](https://github.com/inspektor-gadget/inspektor-gadget/pull/1305).

Reply:
> I did some experiments to make the `runner` goal clearer [here](https://github.com/inspektor-gadget/inspektor-gadget/pull/1305).

I like that very much! And seeing it that way, `GadgetContext` makes total sense.

The only thing that worries me a bit is the return value being fixed to ([]byte, error). In another comment, we concluded that it might as well be ([]*T, error) - so we might need a little more flexibility there.

Reply:
> I like that very much! And seeing it that way, GadgetContext makes totally sense.
>
Great! I also like it! 

> The only thing that worries me a bit is the return value being fixed to ([]byte, error). In another comment, we concluded that it might as well be ([]*T, error) - so we might need a little more flexibility there.
>
I propose to go ahead and review/merge it to avoid it going in conflict with other parallel developments (as changing the name changes many files). Then, we can work on the `[]*T` instead of `[]byte`. WDYT?

Reply:
I agree to merge that soon. It helps reviewing this PR.